### ISSUE TITLE
Keyboard user cannot select 'indeterminate' state on checkbox component

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#1469]
## Environment
Chrome 70
macOS Mojave

## Violation
There are three states to the checkboxes. true(checked), false(unchecked) or mixed (indeterminate). A keyboard user cannot cycle to the indeterminate state using the either spacebar or enter.

**Comment by dakahn** [comment]
Spoke with @alisonjoseph and this is expected/intended behavior and not a bug. 🤦🏼‍♂️ My bad. Closing!
